### PR TITLE
"Report a problem" now validates email and obfuscates it in the issue

### PR DESCRIPTION
**Pull request by hatton**

### CODE REVIEW

File: src/BloomExe/MiscUI/ProblemReporterDialog.cs [hunk 1]
```diff
@@ -148,9 +150,22 @@ private void UpdateDisplay(object sender, EventArgs e)
 #endif
 		}
 
+		public bool IsLegalEmail(string emailAddress)
```
Comment (JohnThomson) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: We have something similar IIRC in the (C#) dialog for signing up for bloom library. Could we share?

File: src/BloomExe/MiscUI/ProblemReporterDialog.cs [hunk 2]
```diff
@@ -148,9 +150,22 @@ private void UpdateDisplay(object sender, EventArgs e)
...
+
 		protected virtual void UpdateDisplay()
 		{
-			_submitButton.Enabled = !string.IsNullOrWhiteSpace(_name.Text.Trim()) && !string.IsNullOrWhiteSpace(_email.Text.Trim()) &&
+			if(!string.IsNullOrWhiteSpace(_email.Text.Trim()))
+			{
+				_email.ForeColor = IsLegalEmail(_email.Text) ? Color.Black : Color.Red;
+			}
+
+			_submitButton.Enabled = !string.IsNullOrWhiteSpace(_name.Text.Trim()) && !string.IsNullOrWhiteSpace(_email.Text.Trim()) && IsLegalEmail(_email.Text) &&
```
Comment (JohnThomson) [inline src/BloomExe/MiscUI/ProblemReporterDialog.cs]: Do we definitely not want a problem report from a rice farmer who doesn't have his own email? I suppose it's unlikely to have internet and not email...seemed worth asking, though.

Will it be sufficiently obvious to a rice farmer that he has to fill in all three fields before submit is enabled? It's good to show email in red if invalid, but I prefer the design where compulsory fields are highlighted in some way when the (not disabled) submit button is clicked.

**Comment by hatton** [comment]
Good points, I'll do something there.

> Do we definitely not want a problem report from a rice farmer who doesn't have his own email?

He doesn't speak a language we can read, either :-) RF is not the user of this feature.

**Comment by hatton** [comment]
OK, decided it's not worth the time here to show which of the 3 boxes you have left blank.

**Pull request merged**